src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs

```diff
@@ -22,4 +22,19 @@ public static class SiteServiceExtensions
     /// <returns>An instance of the given type if one exists.</returns>
     public static async Task<T> GetSettingsAsync<T>(this ISiteService siteService, string name) where T : new()
         => (await siteService.GetSiteSettingsAsync()).As<T>(name);
+
+    /// <summary>
+    /// Gets the site settings from the cache.
+    /// </summary>
+    public static ISite GetSiteSettings(this ISiteService siteService)
+        => siteService.GetSiteSettingsAsync().GetAwaiter().GetResult();
+
+    /// <summary>
+    /// Gets an instance of the specified settings if it exists.
+    /// </summary>
+    /// <typeparam name="T">The type of the settings to attempt to get.</typeparam>
+    /// <param name="siteService">The site service.</param>
+    /// <returns>An instance of the given type if one exists.</returns>
+    public static T GetSettings<T>(this ISiteService siteService) where T : new()
```

Comment:
I think we had extension like this that was removed I believe.

Reply:
I don't see anything related to `ISiteService`, this has always been asynchronous. And the extension methods were added last year by yourself.

Reply:
I think we had one at some point and we removed it so we can use async.